.github/workflows/maven.yml

```diff
@@ -14,7 +14,7 @@ jobs:
     if: github.event_name != 'pull_request' || github.event.pull_request.base.repo.clone_url != github.event.pull_request.head.repo.clone_url
     strategy:
       matrix:
-        os: ['ubuntu-22.04', 'macos-12', 'windows-2022']
+        os: ['ubuntu-22.04', 'macos-14', 'windows-2022']
```

Comment:
Given that we also have cocoa x86_64 in our env, should we still run against macos-12?

Reply:
@szarnekow we could, but we'd lose the advantage of faster macOS environment in GitHub Actions ;)

Reply:
> True. It would be even faster if we skip all environments, no? Kidding.

And we would even have less releng/CI troubles! ;)

> 
> Question is: What do we aim for? Short build times or more coverage?

I see your point. I can also put back the old macOS x86 configuration. We can then decide whether to ditch one if it becomes too flaky